File: VB/DXExample.Module/Updater.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.ExpressApp.Updating
Imports DevExpress.Xpo
Imports DevExpress.Data.Filtering
Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.BaseImpl

Namespace DXExample.Module
	Public Class Updater
		Inherits ModuleUpdater
		Public Sub New(ByVal objectSpace As IObjectSpace, ByVal currentDBVersion As Version)
			MyBase.New(objectSpace, currentDBVersion)
		End Sub
		Public Overrides Sub UpdateDatabaseAfterUpdateSchema()
			MyBase.UpdateDatabaseAfterUpdateSchema()
			Dim testDO1 As DomainObject1 = ObjectSpace.FindObject(Of DomainObject1)(New BinaryOperator("Name", "Test"))
			If testDO1 Is Nothing Then
				testDO1 = ObjectSpace.CreateObject(Of DomainObject1)()
				testDO1.Name = "Test"
				testDO1.Save()
			End If
		End Sub
	End Class
End Namespace

File: CS/DXExample.Module/Updater.cs
using System;

using DevExpress.ExpressApp.Updating;
using DevExpress.Xpo;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.BaseImpl;

namespace DXExample.Module {
    public class Updater : ModuleUpdater {
        public Updater(Session session, Version currentDBVersion) : base(session, currentDBVersion) { }
        public override void UpdateDatabaseAfterUpdateSchema() {
            base.UpdateDatabaseAfterUpdateSchema();
            DomainObject1 testDO1 = Session.FindObject<DomainObject1>(new BinaryOperator("Name", "Test"));
            if (testDO1 == null) {
                testDO1 = new DomainObject1(Session);
                testDO1.Name = "Test";
                testDO1.Save();
            }
        }
    }
}
